zkevm-circuits/src/evm_circuit/execution/error_oog_pure_memory.rs

```diff
@@ -21,9 +21,14 @@ pub(crate) struct ErrorOOGPureMemoryGadget<F> {
     opcode: Cell<F>,
     address: Word<F>,
     address_in_range: IsZeroGadget<F>,
-    memory_expansion:
-        MemoryExpansionGadget<F, { MAX_MEMORY_SIZE_IN_BYTES * 2 - 1 }>,
-    insufficient_gas: LtGadget<F, { MAX_MEMORY_SIZE_IN_BYTES * 2 - 1 }>,
+    // Allow memory size to expand to 5 bytes, because memory address could be
+    // at most 2^40 - 1, after constant division by 32, the memory size then
+    // could be at most 2^35 - 1.
+    memory_expansion: MemoryExpansionGadget<F, 1, { N_BYTES_MEMORY_SIZE + 1 }>,
```

Comment:
The `N_BYTES_MEMORY_SIZE` is `4`, and it's useful in successful cases. But in such OOG case, because we actually use `5` bytes for memory address, after division of `32`, it still could be `5` bytes, so here it's `4 + 1 = 5`.

Reply:
Now it makes sense! Thanks!
Maybe we can declare a const for the OOG method or ad a comment explaining the `+1`

Reply:
Added in [`3909ef6`](https://github.com/appliedzkp/zkevm-circuits/pull/279/commits/3909ef67a4d14f00621818534b4bb25468b6d55f).

Reply:
Ohhh I see. Maybe then just add a small explanation or a new constant for the OOG case would be sufficient :)